crashpad build failure

### Operating system

Windows

### Compiler

MSVC

### Steps to reproduce the behavior

```Shell
MSVC cannot link properly to crashpad when in any release build when using cmake and vcpkg.

-------CMakePresets.json------
{
  "version": 4,
  "cmakeMinimumRequired": {
    "major": 3,
    "minor": 19,
    "patch": 0
  },
  "configurePresets": [
    {
      "name": "windows-release",
      "generator": "Visual Studio 17 2022",
      "binaryDir": "${sourceDir}/out/build/windows-release",
      "cacheVariables": {
        "CMAKE_BUILD_TYPE": "Release",
        "VCPKG_TARGET_TRIPLET": "x64-windows",
        "CMAKE_TOOLCHAIN_FILE": "${sourceDir}/vcpkg/scripts/buildsystems/vcpkg.cmake"
      }
    }
  ],
  "buildPresets": [
    {
      "name": "windows-release-build",
      "configurePreset": "windows-release"
    }
  ]
}

-----CMakeLists.txt ----------------------
cmake_minimum_required(VERSION 3.15)

# Define your project
project(CrashpadDemo)

# Use vcpkg's toolchain file
set(CMAKE_TOOLCHAIN_FILE "${CMAKE_SOURCE_DIR}/vcpkg/scripts/buildsystems/vcpkg.cmake")

# Set C++ standard
set(CMAKE_CXX_STANDARD 17)
set(CMAKE_CXX_STANDARD_REQUIRED ON)

# Add the executable
add_executable(CrashpadDemo main.cpp)

# Link Crashpad libraries
find_package(crashpad CONFIG REQUIRED)
target_link_libraries(CrashpadDemo PRIVATE crashpad::crashpad)

------------vcpkg.json -------------------------
{
    "name": "crashpad-demo",
    "version": "1.0.0",
    "dependencies": [
      "crashpad"
    ]
  }
  

----------main.cpp------------------------------
#include <crashpad/client/crashpad_client.h>
#include <crashpad/client/crashpad_info.h>
#include <iostream>

void causeCrash() {
    int* ptr = nullptr;
    *ptr = 42; // This will crash.
}

int main() {
    // Initialize Crashpad
    // Path where crashpad stores crash reports before uploading.
    base::FilePath db_path(L"Crashpad");

    // Path to the Crashpad handler executable.
    base::FilePath handler_path(L"crashpad_handler.exe");

    // URL to your Crashpad server.
    std::string server_url = "";
    std::vector<std::string> arguments;

    crashpad::CrashpadClient client;

    if (!client.StartHandler(handler_path,
                             db_path,
                             db_path,
                             server_url,
                             {},
                             {},
                             true,
                             true)) {
        std::cerr << "Failed to start Crashpad handler." << std::endl;
        return 1;
    }

    crashpad::CrashpadInfo* crashpad_info = crashpad::CrashpadInfo::GetCrashpadInfo();
    crashpad_info->set_crashpad_handler_behavior(crashpad::TriState::kEnabled);

    std::cout << "Crashpad is initialized. Crashing now..." << std::endl;

    causeCrash(); // Cause a crash

    return 0;
}
```


### Failure logs

PS C:\coding\code_for_fun\crashpad_vcpkg_demo> cmake --preset windows-release
Preset CMake variables:

  CMAKE_BUILD_TYPE="Release"
  CMAKE_TOOLCHAIN_FILE="C:/coding/code_for_fun/crashpad_vcpkg_demo/vcpkg/scripts/buildsystems/vcpkg.cmake"
  VCPKG_TARGET_TRIPLET="x64-windows"

-- Running vcpkg install
Detecting compiler hash for triplet x64-windows...
Compiler found: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.40.33807/bin/Hostx64/x64/cl.exe
The following packages will be built and installed:
    crashpad:x64-windows@2024-04-11#1
  * curl[core,non-http,schannel,ssl,sspi]:x64-windows@8.9.1
  * vcpkg-cmake:x64-windows@2024-04-23
  * vcpkg-cmake-config:x64-windows@2024-05-23
  * vcpkg-cmake-get-vars:x64-windows@2023-12-31
  * vcpkg-gn:x64-windows@2024-02-22
  * vcpkg-tool-gn:x64-windows@2024-02-22
  * zlib:x64-windows@1.3.1
Additional packages (*) will be modified to complete this operation.
Restored 8 package(s) from C:\Users\Chad\AppData\Local\vcpkg\archives in 369 ms. Use --debug to see more details.
Installing 1/8 vcpkg-cmake:x64-windows@2024-04-23...
Elapsed time to handle vcpkg-cmake:x64-windows: 3.34 ms
vcpkg-cmake:x64-windows package ABI: 2a0f6a2d855a4164e32eb75c7376fe6bf9ea3485d9809b29a0ec9c23c5c3e3d1
Installing 2/8 zlib:x64-windows@1.3.1...
Elapsed time to handle zlib:x64-windows: 5.03 ms
zlib:x64-windows package ABI: 42a57d49b5f89e1d4006374fa290d6f2eb01ca7b1c8bab9cc99a2eaf59f1eb43
Installing 3/8 vcpkg-tool-gn:x64-windows@2024-02-22...
Elapsed time to handle vcpkg-tool-gn:x64-windows: 3.52 ms
vcpkg-tool-gn:x64-windows package ABI: b4134a2bad645ed564894cf1c6ec891f446ed5517eb7d1547aa1d97a6bedff81
Installing 4/8 vcpkg-gn:x64-windows@2024-02-22...
Elapsed time to handle vcpkg-gn:x64-windows: 2.7 ms
vcpkg-gn:x64-windows package ABI: 4057ff7e28626fc8aacda2a80f7eaeb23484a2f46ba222a4e818b6bb17314db3
Installing 5/8 vcpkg-cmake-get-vars:x64-windows@2023-12-31...
Elapsed time to handle vcpkg-cmake-get-vars:x64-windows: 3.09 ms
vcpkg-cmake-get-vars:x64-windows package ABI: ce2ed716e1677d65b4e6a940f2b58cd4f95f2f126202da24d83bf9ae5b55deca
Installing 6/8 vcpkg-cmake-config:x64-windows@2024-05-23...
Elapsed time to handle vcpkg-cmake-config:x64-windows: 2.71 ms
vcpkg-cmake-config:x64-windows package ABI: 32421728ca7fca59ac93ffc023211a23962e31933dfb3504f70e95d2449d19cc
Installing 7/8 curl[core,non-http,schannel,ssl,sspi]:x64-windows@8.9.1...
Elapsed time to handle curl:x64-windows: 11 ms
curl:x64-windows package ABI: 9e2bcf9f5f18d35cd0e0e3ba27e42d2d21a673d35d9d5c5961cb92d60a24afb3
Installing 8/8 crashpad:x64-windows@2024-04-11#1...
Elapsed time to handle crashpad:x64-windows: 48.2 ms
crashpad:x64-windows package ABI: 968b1ecce3d534294dc41874ad1baef4b3efb0704ea31d27819144a289d9259e
Total install time: 79.7 ms
crashpad provides CMake targets:

  # this is heuristically generated, and may not be correct
  find_package(crashpad CONFIG REQUIRED)
  target_link_libraries(main PRIVATE crashpad::crashpad)

-- Running vcpkg install - done
-- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.22631.
-- The C compiler identification is MSVC 19.40.33813.0
-- The CXX compiler identification is MSVC 19.40.33813.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.40.33807/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.40.33807/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found ZLIB: optimized;C:/coding/code_for_fun/crashpad_vcpkg_demo/out/build/windows-release/vcpkg_installed/x64-windows/lib/zlib.lib;debug;C:/coding/code_for_fun/crashpad_vcpkg_demo/out/build/windows-release/vcpkg_installed/x64-windows/debug/lib/zlibd.lib (found version "1.3.1")  
-- Configuring done (4.5s)
-- Generating done (0.0s)
-- Build files have been written to: C:/coding/code_for_fun/crashpad_vcpkg_demo/out/build/windows-release
PS C:\coding\code_for_fun\crashpad_vcpkg_demo> cmake --build --preset windows-release-build
MSBuild version 17.10.4+10fbfbf2e for .NET Framework

  1>Checking Build System
  Building Custom Rule C:/coding/code_for_fun/crashpad_vcpkg_demo/CMakeLists.txt
  main.cpp
common.lib(common.crashpad_info.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
common.lib(common.crashpad_info.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
client.lib(client.crashpad_client_win.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
client.lib(client.crashpad_client_win.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.scoped_handle.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.scoped_handle.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.random_string.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.random_string.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.command_line.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.command_line.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.critical_section_with_debug_info.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.critical_section_with_debug_info.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.get_function.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.get_function.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.handle.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.handle.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.initial_client_data.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.initial_client_data.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.ntstatus_logging.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.ntstatus_logging.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.process_info.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.process_info.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.registration_protocol_win.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.registration_protocol_win.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.scoped_process_suspend.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.scoped_process_suspend.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.file_io.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.file_io.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.string_number_conversion.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.string_number_conversion.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.split_string.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.split_string.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.aligned_allocator.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.aligned_allocator.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.scoped_local_alloc.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.scoped_local_alloc.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.file_io_win.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
util.lib(util.file_io_win.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
base.lib(base.file_path.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
base.lib(base.file_path.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
base.lib(base.logging.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
base.lib(base.logging.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
base.lib(base.stringprintf.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
base.lib(base.stringprintf.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
base.lib(base.utf_string_conversions.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
base.lib(base.utf_string_conversions.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
base.lib(base.rand_util.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
base.lib(base.rand_util.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
base.lib(base.string_util_win.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
base.lib(base.string_util_win.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
base.lib(base.utf_string_conversion_utils.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
base.lib(base.utf_string_conversion_utils.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in main.obj [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
LINK : warning LNK4098: defaultlib 'MSVCRT' conflicts with use of other libs; use /NODEFAULTLIB:library [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\Debug\CrashpadDemo.exe : fatal error LNK1319: 52 mismatches detected [C:\coding\code_for_fun\crashpad_vcpkg_demo\out\build\windows-release\CrashpadDemo.vcxproj]
PS C:\coding\code_for_fun\crashpad_vcpkg_demo> 

### Additional context

You can manually change the link lib in the vcxproj file to the non debug versions and a project will compile properly.

--- a/ports/crashpad/crashpadConfig.cmake.in
+++ b/ports/crashpad/crashpadConfig.cmake.in
@@ -38,6 +38,10 @@ if(NOT TARGET crashpad::crashpad)
 
   target_link_libraries(crashpad::crashpad INTERFACE ZLIB::ZLIB)
 
+  if(NOT TARGET crashpad)
+    add_library(crashpad ALIAS crashpad::crashpad)
+  endif()
+
   unset(_lib)
   unset(_libs)
 endif()

--- a/ports/crashpad/crashpadConfig.cmake.in
+++ b/ports/crashpad/crashpadConfig.cmake.in
@@ -6,26 +6,35 @@ if(_IMPORT_PREFIX STREQUAL "/")
   set(_IMPORT_PREFIX "")
 endif()
 
-add_library(crashpad INTERFACE)
-add_library(crashpad::crashpad ALIAS crashpad)
+include(CMakeFindDependencyMacro)
+find_dependency(ZLIB)
 
-set(CRASHPAD_LIBRARIES common client util base)
+if(NOT TARGET crashpad::crashpad)
+  add_library(crashpad::crashpad INTERFACE IMPORTED)
+  target_include_directories(crashpad::crashpad INTERFACE "${_IMPORT_PREFIX}/include/crashpad")
 
-if(WIN32)
-	target_compile_definitions(crashpad INTERFACE NOMINMAX)
-elseif(APPLE)
-	list(APPEND CRASHPAD_LIBRARIES ApplicationServices
-        CoreFoundation Foundation IOKit Security bsm mig_output)
+  set(_libs client common util base)
+  if(APPLE)
+    list(APPEND _libs mig_output)
+  endif()
+  include(SelectLibraryConfigurations)
+  foreach(_lib IN LISTS _libs)
+    find_library(CRASHPAD_${_lib}_LIBRARY_RELEASE NAMES ${_lib} PATHS "${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/lib" NO_DEFAULT_PATH)
+    find_library(CRASHPAD_${_lib}_LIBRARY_DEBUG NAMES ${_lib} PATHS "${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/debug/lib" NO_DEFAULT_PATH)
+    select_library_configurations(CRASHPAD_${_lib})
+    target_link_libraries(crashpad::crashpad INTERFACE ${CRASHPAD_${_lib}_LIBRARIES})
+  endforeach()
+
+  if(WIN32)
+  	target_compile_definitions(crashpad::crashpad INTERFACE NOMINMAX)
+  elseif(APPLE)
+    foreach(_lib IN ITEMS ApplicationServices CoreFoundation Foundation IOKit Security bsm)
+      find_library(CRASHPAD_${_lib}_LIBRARY NAMES ${_lib})
+      target_link_libraries(crashpad::crashpad INTERFACE ${CRASHPAD_${_lib}_LIBRARY})
+    endforeach()
+  endif()
+
+  target_link_libraries(crashpad::crashpad INTERFACE ZLIB::ZLIB)
+
+  unset(_lib)
 endif()
-
-foreach(LIB_NAME ${CRASHPAD_LIBRARIES})
-  find_library(_LIB ${LIB_NAME})
-  target_link_libraries(crashpad INTERFACE ${_LIB})
-  unset(_LIB CACHE)
-endforeach()
-
-find_package(ZLIB REQUIRED)
-target_link_libraries(crashpad INTERFACE ZLIB::ZLIB)
-
-target_include_directories(crashpad 
-        INTERFACE ${_IMPORT_PREFIX}/include/crashpad)

--- a/ports/crashpad/crashpadConfig.cmake.in
+++ b/ports/crashpad/crashpadConfig.cmake.in
@@ -37,4 +37,5 @@ if(NOT TARGET crashpad::crashpad)
   target_link_libraries(crashpad::crashpad INTERFACE ZLIB::ZLIB)
 
   unset(_lib)
+  unset(_libs)
 endif()

--- a/ports/crashpad/crashpadConfig.cmake.in
+++ b/ports/crashpad/crashpadConfig.cmake.in
@@ -32,6 +32,8 @@ if(NOT TARGET crashpad::crashpad)
       find_library(CRASHPAD_${_lib}_LIBRARY NAMES ${_lib})
       target_link_libraries(crashpad::crashpad INTERFACE ${CRASHPAD_${_lib}_LIBRARY})
     endforeach()
+  elseif(ANDROID)
+    target_link_libraries(crashpad::crashpad INTERFACE log)
   endif()
 
   target_link_libraries(crashpad::crashpad INTERFACE ZLIB::ZLIB)

--- a/ports/crashpad/ndk-toolchain.diff
+++ b/ports/crashpad/ndk-toolchain.diff
@@ -1,0 +1,22 @@
+diff --git a/build/config/BUILD.gn b/build/config/BUILD.gn
+index 0802be8..dd77285 100644
+--- a/build/config/BUILD.gn
++++ b/build/config/BUILD.gn
+@@ -425,7 +425,7 @@ toolchain("gcc_like_toolchain") {
+                               ])
+ 
+     if (target_cpu == "arm") {
+-      tool_prefix = "arm-linux-androideabi"
++      # unused
+     } else if (target_cpu == "arm64") {
+       tool_prefix = "aarch64-linux-android"
+     } else if (target_cpu == "x86") {
+@@ -446,7 +446,7 @@ toolchain("gcc_like_toolchain") {
+     cxx = ndk_bin_dir + clang_prefix + "-clang++"
+     asm = cxx
+     ld = cxx
+-    ar = ndk_bin_dir + tool_prefix + "-ar"
++    ar = ndk_bin_dir + "llvm-ar"
+   } else {
+     cc = "clang"
+     cxx = "clang++"

--- a/ports/crashpad/portfile.cmake
+++ b/ports/crashpad/portfile.cmake
@@ -11,40 +11,26 @@ vcpkg_from_git(
 vcpkg_find_acquire_program(PYTHON3)
 vcpkg_replace_string("${SOURCE_PATH}/.gn" "script_executable = \"python3\"" "script_executable = \"${PYTHON3}\"")
 
-function(checkout_in_path PATH URL REF)
-    if(EXISTS "${PATH}")
-        return()
-    endif()
-
-    vcpkg_from_git(
-        OUT_SOURCE_PATH DEP_SOURCE_PATH
-        URL "${URL}"
-        REF "${REF}"
-    )
-    file(RENAME "${DEP_SOURCE_PATH}" "${PATH}")
-    file(REMOVE_RECURSE "${DEP_SOURCE_PATH}")
-endfunction()
-
 # mini_chromium contains the toolchains and build configuration
-checkout_in_path(
-    "${SOURCE_PATH}/third_party/mini_chromium/mini_chromium"
-    "https://chromium.googlesource.com/chromium/mini_chromium"
-    "dce72d97d1c2e9beb5e206c6a05a702269794ca3"
-)
-vcpkg_apply_patches(
-    SOURCE_PATH "${SOURCE_PATH}/third_party/mini_chromium/mini_chromium"
-    PATCHES
-      fix-std-20.patch
-      ndk-toolchain.diff
-)      
-
-if(VCPKG_TARGET_IS_ANDROID OR VCPKG_TARGET_IS_LINUX)
-    # fetch lss
-    checkout_in_path(
-        "${SOURCE_PATH}/third_party/lss/lss"
-        https://chromium.googlesource.com/linux-syscall-support
-        9719c1e1e676814c456b55f5f070eabad6709d31
+if(NOT EXISTS "${SOURCE_PATH}/third_party/mini_chromium/mini_chromium/BUILD.gn")
+    vcpkg_from_git(OUT_SOURCE_PATH mini_chromium
+        URL "https://chromium.googlesource.com/chromium/mini_chromium"
+        REF dce72d97d1c2e9beb5e206c6a05a702269794ca3
+        PATCHES
+            fix-std-20.patch
+            ndk-toolchain.diff
     )
+    file(REMOVE_RECURSE "${SOURCE_PATH}/third_party/mini_chromium/mini_chromium")
+    file(RENAME "${mini_chromium}" "${SOURCE_PATH}/third_party/mini_chromium/mini_chromium")
+endif()
+
+if(NOT EXISTS "${SOURCE_PATH}/third_party/lss/lss/BUILD.gn" AND (VCPKG_TARGET_IS_ANDROID OR VCPKG_TARGET_IS_LINUX))
+    vcpkg_from_git(OUT_SOURCE_PATH lss
+        URL https://chromium.googlesource.com/linux-syscall-support
+        REF 9719c1e1e676814c456b55f5f070eabad6709d31
+    )
+    file(REMOVE_RECURSE "${SOURCE_PATH}/third_party/lss/lss")
+    file(RENAME "${lss}" "${SOURCE_PATH}/third_party/lss/lss")
 endif()
 
 function(replace_gn_dependency INPUT_FILE OUTPUT_FILE LIBRARY_NAMES)

--- a/ports/crashpad/portfile.cmake
+++ b/ports/crashpad/portfile.cmake
@@ -68,7 +68,20 @@ set(OPTIONS "target_cpu=\"${VCPKG_TARGET_ARCHITECTURE}\"")
 set(OPTIONS_DBG "is_debug=true")
 set(OPTIONS_REL "")
 
-if(CMAKE_HOST_WIN32)
+if(VCPKG_TARGET_IS_ANDROID)
+    vcpkg_cmake_get_vars(cmake_vars_file)
+    include("${cmake_vars_file}")
+    string(APPEND OPTIONS " target_os=\"android\" android_ndk_root=\"${VCPKG_DETECTED_CMAKE_ANDROID_NDK}\"")
+
+elseif(VCPKG_TARGET_IS_LINUX)
+    string(APPEND OPTIONS " target_os=\"linux\"")
+
+elseif(VCPKG_TARGET_IS_OSX)
+    string(APPEND OPTIONS " target_os=\"mac\"")
+
+elseif(VCPKG_TARGET_IS_WINDOWS AND NOT VCPKG_TARGET_IS_MINGW)
+    string(APPEND OPTIONS " target_os=\"win\"")
+
     # Load toolchains
     vcpkg_cmake_get_vars(cmake_vars_file)
     include("${cmake_vars_file}")
@@ -92,12 +105,6 @@ if(CMAKE_HOST_WIN32)
 
     set(OPTIONS_DBG "${OPTIONS_DBG} ${DISABLE_WHOLE_PROGRAM_OPTIMIZATION}")
     set(OPTIONS_REL "${OPTIONS_REL} ${DISABLE_WHOLE_PROGRAM_OPTIMIZATION}")
-endif()
-
-if(VCPKG_TARGET_IS_ANDROID)
-    vcpkg_cmake_get_vars(cmake_vars_file)
-    include("${cmake_vars_file}")
-    string(APPEND OPTIONS " target_os=\"android\" android_ndk_root=\"${VCPKG_DETECTED_CMAKE_ANDROID_NDK}\"")
 endif()
 
 vcpkg_gn_configure(

--- a/ports/crashpad/portfile.cmake
+++ b/ports/crashpad/portfile.cmake
@@ -34,10 +34,11 @@ checkout_in_path(
 vcpkg_apply_patches(
     SOURCE_PATH "${SOURCE_PATH}/third_party/mini_chromium/mini_chromium"
     PATCHES
-      "fix-std-20.patch"
+      fix-std-20.patch
+      ndk-toolchain.diff
 )      
 
-if(VCPKG_TARGET_IS_LINUX)
+if(VCPKG_TARGET_IS_ANDROID OR VCPKG_TARGET_IS_LINUX)
     # fetch lss
     checkout_in_path(
         "${SOURCE_PATH}/third_party/lss/lss"
@@ -77,6 +78,7 @@ replace_gn_dependency(
     "z;zlib;zlibd"
 )
 
+set(OPTIONS "target_cpu=\"${VCPKG_TARGET_ARCHITECTURE}\"")
 set(OPTIONS_DBG "is_debug=true")
 set(OPTIONS_REL "")
 
@@ -106,9 +108,15 @@ if(CMAKE_HOST_WIN32)
     set(OPTIONS_REL "${OPTIONS_REL} ${DISABLE_WHOLE_PROGRAM_OPTIMIZATION}")
 endif()
 
+if(VCPKG_TARGET_IS_ANDROID)
+    vcpkg_cmake_get_vars(cmake_vars_file)
+    include("${cmake_vars_file}")
+    string(APPEND OPTIONS " target_os=\"android\" android_ndk_root=\"${VCPKG_DETECTED_CMAKE_ANDROID_NDK}\"")
+endif()
+
 vcpkg_gn_configure(
     SOURCE_PATH "${SOURCE_PATH}"
-    OPTIONS " target_cpu=\"${VCPKG_TARGET_ARCHITECTURE}\" "
+    OPTIONS "${OPTIONS}"
     OPTIONS_DEBUG "${OPTIONS_DBG}"
     OPTIONS_RELEASE "${OPTIONS_REL}"
 )

--- a/ports/crashpad/vcpkg.json
+++ b/ports/crashpad/vcpkg.json
@@ -10,7 +10,11 @@
   "license": "Apache-2.0",
   "supports": "linux | osx | (windows & !uwp)",
   "dependencies": [
-    "curl",
+    {
+      "name": "curl",
+      "default-features": false,
+      "platform": "linux"
+    },
     {
       "name": "vcpkg-cmake-get-vars",
       "host": true

--- a/ports/crashpad/vcpkg.json
+++ b/ports/crashpad/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "crashpad",
   "version-date": "2024-04-11",
-  "port-version": 1,
+  "port-version": 2,
   "description": [
     "Crashpad is a crash-reporting system.",
     "Crashpad is a library for capturing, storing and transmitting postmortem crash reports from a client to an upstream collection server. Crashpad aims to make it possible for clients to capture process state at the time of crash with the best possible fidelity and coverage, with the minimum of fuss."

--- a/ports/crashpad/vcpkg.json
+++ b/ports/crashpad/vcpkg.json
@@ -8,7 +8,7 @@
   ],
   "homepage": "https://chromium.googlesource.com/crashpad/crashpad/+/master/README.md",
   "license": "Apache-2.0",
-  "supports": "linux | osx | (windows & !uwp)",
+  "supports": "android | linux | osx | (windows & !uwp)",
   "dependencies": [
     {
       "name": "curl",

--- a/scripts/test_ports/vcpkg-ci-crashpad/portfile.cmake
+++ b/scripts/test_ports/vcpkg-ci-crashpad/portfile.cmake
@@ -1,0 +1,4 @@
+set(VCPKG_POLICY_EMPTY_PACKAGE enabled)
+
+vcpkg_cmake_configure(SOURCE_PATH "${CURRENT_PORT_DIR}/project")
+vcpkg_cmake_build()

--- a/scripts/test_ports/vcpkg-ci-crashpad/project/CMakeLists.txt
+++ b/scripts/test_ports/vcpkg-ci-crashpad/project/CMakeLists.txt
@@ -1,0 +1,9 @@
+cmake_minimum_required(VERSION 3.29)
+project(crashpad-test CXX)
+
+set(CMAKE_CXX_STANDARD 11)
+
+add_executable(main main.cpp)
+
+find_package(crashpad CONFIG REQUIRED)
+target_link_libraries(main PRIVATE crashpad::crashpad)

--- a/scripts/test_ports/vcpkg-ci-crashpad/project/main.cpp
+++ b/scripts/test_ports/vcpkg-ci-crashpad/project/main.cpp
@@ -1,0 +1,7 @@
+#include "client/crashpad_client.h"
+
+int main()
+{
+    auto *client = new crashpad::CrashpadClient();
+    return 0;
+}

--- a/scripts/test_ports/vcpkg-ci-crashpad/vcpkg.json
+++ b/scripts/test_ports/vcpkg-ci-crashpad/vcpkg.json
@@ -1,0 +1,12 @@
+{
+  "name": "vcpkg-ci-crashpad",
+  "version-string": "ci",
+  "description": "Validates crashpad",
+  "dependencies": [
+    "crashpad",
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    }
+  ]
+}

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2014,7 +2014,7 @@
     },
     "crashpad": {
       "baseline": "2024-04-11",
-      "port-version": 1
+      "port-version": 2
     },
     "crashrpt": {
       "baseline": "1.4.3",

--- a/versions/c-/crashpad.json
+++ b/versions/c-/crashpad.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "5ef5b1ce847e728620bbb612b3fc2d6ba5bf129f",
+      "version-date": "2024-04-11",
+      "port-version": 2
+    },
+    {
       "git-tree": "43c2573b9400b543b26e7c9fbb091bdfefa51b79",
       "version-date": "2024-04-11",
       "port-version": 1

--- a/versions/c-/crashpad.json
+++ b/versions/c-/crashpad.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "5ef5b1ce847e728620bbb612b3fc2d6ba5bf129f",
+      "git-tree": "a8c3dac7327cb12c8f1885797a7d63ed99b6963c",
       "version-date": "2024-04-11",
       "port-version": 2
     },

